feat: add HPA for hydra +  mockTlsTermination for example-idp

pommelinho opened the pull request:
## Related issue

<!--
Please link the GitHub issue this pull request resolves in the format of `#1234`. If you discussed this change
with a maintainer, please mention her/him using the `@` syntax (e.g. `@aeneasr`).

If this change neither resolves an existing issue nor has sign-off from one of the maintainers, there is a
chance substantial changes will be requested or that the changes will be rejected.

You can discuss changes with maintainers either in the [ORY Community Forums](https://community.ory.sh/) or
join the [ORY Chat](https://www.ory.sh/chat).
-->

## Proposed changes

<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
-->

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

## Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution
you did and what alternatives you considered, etc...
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ory/k8s?pullRequest=197) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ory/k8s?pullRequest=197) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ory/k8s?pullRequest=197) it.</sub>

pommelinho commented:
Added HPA for hydra chart.This is partially related to #194 .
Additionally i added support env MOCK_TLS_TERMINATION in example-idp chart which might be useful so that exampl-idp can communicae with hydra admin api.

aeneasr commented on helm/charts/hydra/values.yaml:
```diff
@@ -163,6 +163,14 @@ deployment:
 
+# Configure HPA
+autoscaling:
+  enabled: true
```
Can we move this to the "deployment" top level key? Also let's disable this per default!

pommelinho replied:
sure

aeneasr reviewed (changes requested):
Awesome, thank you for your contribution! This looks pretty good but I have some ideas how to improve it further :)

aeneasr reviewed (approved):
Awesome 🎉 

Thank you for your contribution!
